sdk/ai/azopenaiassistants/client.go

```diff
@@ -0,0 +1,1723 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// Code generated by Microsoft (R) AutoRest Code Generator. DO NOT EDIT.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+package azopenaiassistants
+
+import (
+	"context"
+	"errors"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+)
+
+// Client contains the methods for the OpenAIAssistants group.
+// Don't use this type directly, use a constructor function instead.
+type Client struct {
+	clientData
```

Comment:
Is there a reason this is anonymously embedded? It would be better to not do this. It's too easy to inadvertently change (or add) exported methods on this type without realizing it affects public surface area.

Reply:
That's a good point - it's not embedded for any particular reason. I'll make it a named field instead.